backbone.js

```diff
@@ -346,11 +353,13 @@
       changing        = this._changing;
       this._changing  = true;
 
+      current = this.attributes, prev = this._previousAttributes;
+
       if (!changing) {
-        this._previousAttributes = _.clone(this.attributes);
+        prev = this._previousAttributes = {};
+        for( attr in current ){ prev[ attr ] = current[ attr ]; }
```

Comment:
I'm not a big fan of this either, but I've got a test showing 10x performance improvement in Chrome. I will share it.

In the case of Model.attributes it's just a simple hash, so there are no any need for "hasOwnProperties". So, in this special case this is an opportunity for optimisation. Cause Model.set is slow like hell, because of _.clone and _.isEqual.